### PR TITLE
Report Monitoring products to subscription matcher

### PR DESCRIPTION
**Pull request by hustodemon**
## What does this PR change?
 
 In a similar fashion as the Management products, report Monitoring products
 too. The products are reported based on whether the system has Monitoring
 entitlement assigned.

**Review commit-by-commit** please.

 TODO:
 - [x] Monitoring SUMA instance itself (** Also take into account the ISS slave scenario ** )
 - [x] Adjust the suma server reported products (now we still report 3.1 and sle 12 sp2)
 - [x] ~Monitoring other Prometheus instances (I don't know what they are and how to
 detect them)~
 - [x] Fix the formula assignment bug in `SystemManager.entitleServer`
 - [x] In `Tests for Monitoring products reporting ` commit: there are 2 todos (re-assignment of server arch to be able to assign the Monitoring entitlement (it failed when the server arch was `i386`))
- [x] Answer this: Do we want to report Monitoring product even when no base product is installed on the machine (currently it's not reported in such case). Let's keep it that way.
 
 ## GUI diff
 
 No difference.
 
 - [x] **DONE**
 
 ## Documentation
 - [doc-susemanager](https://github.com/SUSE/doc-susemanager) PR or issue was created (GitHub automatic link expected below)
 
 - [ ] **DONE**
 
 ## Test coverage
 - Unit tests were added
 
 - [x] **DONE**
 
 ## Links
 
 Tracks https://github.com/SUSE/spacewalk/issues/6675
 
 - [x] **DONE**
 
 ## Changelogs
 
 If you don't need a changelog check, please mark this checkbox:
 
 - [ ] No changelog needed
 
 If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)
 
 
 ## Re-run a test
 
 If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:
 
 - [ ] Re-run test "changelog_test"
 - [ ] Re-run test "backend_unittests_pgsql"
 - [ ] Re-run test "java_lint_checkstyle" 		 
 - [ ] Re-run test "java_pgsql_tests"  		 
 - [ ] Re-run test "ruby_rubocop"
 - [ ] Re-run test "schema_migration_test_oracle"
 - [ ] Re-run test "schema_migration_test_pgsql"		 
 - [ ] Re-run test "susemanager_unittests"
 - [ ] Re-run test "javascript_lint"

### CODE REVIEW

**Comment by hustodemon** [comment]
@mcalmer I don't want you to hold you back with the review of the whole PR, but I'd appreciate if you took a quick glance at:
- the latest commit ( https://github.com/uyuni-project/uyuni/pull/1019/commits/a7182cef38c8017ae36e79b15341f9824c37741d) (we need also adjustments for [3.2](https://github.com/SUSE/spacewalk/issues/7899))
- and maybe this: https://github.com/uyuni-project/uyuni/pull/1019/commits/e8d31bdd170415ca4c084c3ffe3a0639b7bf3d61 (we had the same situation for single vs. unlimited virtual **products** for lifecycle subscriptions. i solved it in a same way here)

File: java/code/src/com/suse/manager/matcher/MatcherJsonIO.java [hunk 1]
```diff
@@ -136,10 +136,10 @@ public MatcherJsonIO() {
...
 
         selfProductsByArch = new HashMap<>();
-        // SUSE Manager Server 3.1 x86_64 & SUSE Linux Enterprise Server 12 SP2 x86_64
-        selfProductsByArch.put(AMD64_ARCH_STR, Arrays.asList(1518L, 1357L));
-        // SUSE Manager Server 3.1 s390 & SUSE Linux Enterprise Server 12 SP2 s390
-        selfProductsByArch.put(S390_ARCH_STR, Arrays.asList(1519L, 1356L)); // SUSE Manager Server 3.1 x86_64
+        // SUSE Manager Server 4.0 x86_64 & SUSE Linux Enterprise Server 15 SP1 x86_64
+        selfProductsByArch.put(AMD64_ARCH_STR, Arrays.asList(1899L, 1763L));
+        // SUSE Manager Server 4.0 x86_64 & SUSE Linux Enterprise Server 15 SP1 s390
+        selfProductsByArch.put(S390_ARCH_STR, Arrays.asList(1898L, 1762L));
```
Comment (mcalmer) [inline java/code/src/com/suse/manager/matcher/MatcherJsonIO.java]: Hmm, I see multiple problems, but changing the IDs is makes sense in general. But:

- With 4.0 we are a base product. So we only need one (1) product id. SLES15 SP1 is not needed
- We miss ppc64le. This needs to be added as well.
- The comment for s390 is wrong - you say x86_64

Comment (hustodemon) [reply java/code/src/com/suse/manager/matcher/MatcherJsonIO.java]: Good points. Addressed in the latest commits. Thanks!

**Comment by hustodemon** [comment]
Thanks everyone for the review! I'll proceed with merging.

**Pull request merged**